### PR TITLE
Dice_roll scripting task - task .md file

### PR DESCRIPTION
**Pull request by Aritma**

### CODE REVIEW

File: scripts/tasks/dice_roll/task.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        "rolls" [
+            3,
+            2
+        ]  
+      }  
+  ],  
+  "sum": 16  
+}
+```
+&nbsp;
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:110]: This is unnecessary stretching of the idea with the JSON format,  I suggest we replace a "VERY HARD" assignment with an "DIFFICULT" one as follows:
- enhance the CLI input with an '--num-rolls' option
- create a CSV file tracking all dice rolls with the following structure:
```
DIE N | SIDES | ROLL 1 | ROLL 2 | ROLL N|
```

File: scripts/tasks/dice_roll/task.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+```
+./roll_dice.sh 3d12  
+11  
+7  
+2
+```
+&nbsp;
+
+---
+**AVERAGE:**
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:60]: Rename AVERAGE to MEDIUM

File: scripts/tasks/dice_roll/task.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
+# Dice roll
+
+**Difficulty**: EASY - AVERAGE
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:3]: EASY -> DIFFICULT

File: scripts/tasks/dice_roll/task.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
+# Dice roll
+
+**Difficulty**: EASY - AVERAGE  
+**Linux related topics**: Text processing and filters
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:4]: Scripting is the final topic of the course, so students should be able to figure out what topics are relevant to the solution, so I'd simply drop this line.

File: scripts/tasks/dice_roll/task.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+---
+## Task:
+Create a script which simulates dice roll with different types and numbers of dice.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:8]: "simulates a roll of a die. There can be more than a single die and the dice can be of various shapes (i.e. have more or less than 6 sides)"

File: scripts/tasks/dice_roll/task.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
+# Dice roll
+
+**Difficulty**: EASY - AVERAGE  
+**Linux related topics**: Text processing and filters
+
+---
+## Task:
+Create a script which simulates dice roll with different types and numbers of dice.  
+Follow tasks from number 1 to slowly add functionality into your code.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:9]: Follow the subtasks below starting with number 1, gradually adding functionality to your script.

File: scripts/tasks/dice_roll/task.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+**Linux related topics**: Text processing and filters
+
+---
+## Task:
+Create a script which simulates dice roll with different types and numbers of dice.  
+Follow tasks from number 1 to slowly add functionality into your code.  
+
+For quick reference to specific die we will use NdS notation where N is number of dice  
+and S is number of die sides.  
+e.g. 2d8 means 2 dice with 8 sides
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:13]: This bit of information needs to come before telling students to follow subtasks.

Moreover:
"The notation we're going to use to specify dice/sides is as follows:"
```
NdS where
N - number of dice to be rolled
S - number of sides each of the N dice should have
e.g. 2d8 would denote 2 dice with 8 sides each (octahedron)

File: scripts/tasks/dice_roll/task.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+&nbsp;
+
+---
+**EASY:**
+
+**1) Single 1d6**  
+Create the most basic implementation of one die roll. Roll 1d6.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:21]: Start by implementing a basic single cube die roll (i.e. _1d6_). The script doesn't need to take any argument for this one, in other words, if no argument is given, the script should assume rolling a _1d6_.

File: scripts/tasks/dice_roll/task.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+./roll_dice.sh  
+5
+```
+&nbsp;
+
+**2) Multiple 1d6**  
+Make your script roll Nd6 based on your input.
+
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:31]: The result of this subtask is a bit in conflict with the overall output structure which revolves around the NdS notation, so I suggest you drop this one, it'll be covered by '4) Multiple 1dN' anyway

File: scripts/tasks/dice_roll/task.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+```
+./roll_dice.sh 2  
+3  
+1  
+```
+&nbsp;
+
+**3) Single 1dN. Use die reference in input.**
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:39]: Name this "Roll a single S-sided die. Make your script accept the 1dS argument" please.

File: scripts/tasks/dice_roll/task.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+./roll_dice.sh 2  
+3  
+1  
+```
+&nbsp;
+
+**3) Single 1dN. Use die reference in input.**  
+Make your script roll single 1dN based on your input.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:40]: "roll a single 1dS" (you're not following your own notation ;))

File: scripts/tasks/dice_roll/task.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+./roll_dice.sh 1d12  
+11
+```
+&nbsp;
+
+**4) Multiple 1dN.**
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:48]: "Roll N S-sided dice."

File: scripts/tasks/dice_roll/task.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+Make your script roll single 1dN based on your input.
+
+```
+./roll_dice.sh 1d12  
+11
+```
+&nbsp;
+
+**4) Multiple 1dN.**  
+Make your script roll single NdN based on your input.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:49]: 3d12 in your example doesn't comply with 'NdN' in the subtask ;).
It should IMO read "Make your script roll NdS dice based on the input."

File: scripts/tasks/dice_roll/task.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+11  
+7  
+2
+```
+&nbsp;
+
+---
+**AVERAGE:**
+
+**5) Various combinations of NdN.**
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:62]: "Variation of NdS dice"

File: scripts/tasks/dice_roll/task.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+7  
+2
+```
+&nbsp;
+
+---
+**AVERAGE:**
+
+**5) Various combinations of NdN.**  
+Make your script roll single NdN based on your input.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:63]: "Make your script roll a variation of NdS dice based on the input."

File: scripts/tasks/dice_roll/task.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+```
+./roll_dice.sh 1d12 2d4  
+11  
+3  
+2
+```
+&nbsp;
+
+**6) Add some pretty output**  
+Make your script roll single NdN based on your input. You are not limited in your creativity, add any type of output you like.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:74]: You're only limited by your imagination here, add some pretty output to your roll results :) .

File: scripts/tasks/dice_roll/task.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+./roll_dice.sh 1d12 2d4  
+11  
+3  
+2
+```
+&nbsp;
+
+**6) Add some pretty output**  
+Make your script roll single NdN based on your input. You are not limited in your creativity, add any type of output you like.  
+If you want to make it more complicated, implement option to switch between different output types (human-readable, simple, json format,...)
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:75]: I would leave switching in between formats out, that's quite advanced.

File: scripts/tasks/dice_roll/task.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+            2
+        ]  
+      }  
+  ],  
+  "sum": 16  
+}
+```
+&nbsp;
+
+**7) Input check and error handling**
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:112]: "checking"

File: scripts/tasks/dice_roll/task.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+        ]  
+      }  
+  ],  
+  "sum": 16  
+}
+```
+&nbsp;
+
+**7) Input check and error handling**  
+Make sure your code handles invalid input args.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:113]: "Make sure your script can handle invalid input. Maybe add some nice help on how the script should be used."

File: scripts/tasks/dice_roll/task.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+USE: ./roll_dice NdS
+where N (number of dice) and S (number of sides) are positive non-zero numbers
+```
+&nbsp;
+
+## What to do next?
+If you solved most of defined tasks, you can move to another task or you can improve you solution even more. You can add colors, help option, different output types, export into a file or a permanent logfile which contains all results ever generated by the script with related users and timestamps.
+
+> **ADVICE:**  
+> Don't waste your time to improve this script until you try/finish different tasks.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:134]: "to improve" -> "on improving"
"try/" -> ''
"different" -> "the rest of"

File: scripts/tasks/dice_roll/task.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+where N (number of dice) and S (number of sides) are positive non-zero numbers
+```
+&nbsp;
+
+## What to do next?
+If you solved most of defined tasks, you can move to another task or you can improve you solution even more. You can add colors, help option, different output types, export into a file or a permanent logfile which contains all results ever generated by the script with related users and timestamps.
+
+> **ADVICE:**  
+> Don't waste your time to improve this script until you try/finish different tasks.  
+> Your target should be to try multiple different tasks to see different uses for scripts.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:135]: "The goal should be to tackle as many subtasks as possible to hone your scripting skills gradually."

File: scripts/tasks/dice_roll/task.md [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+```
+&nbsp;
+
+## What to do next?
+If you solved most of defined tasks, you can move to another task or you can improve you solution even more. You can add colors, help option, different output types, export into a file or a permanent logfile which contains all results ever generated by the script with related users and timestamps.
+
+> **ADVICE:**  
+> Don't waste your time to improve this script until you try/finish different tasks.  
+> Your target should be to try multiple different tasks to see different uses for scripts.  
+> You can return to this task anytime later.
```
Comment (eskultety) [inline scripts/tasks/dice_roll/task.md:136]: "You can always return to the unfinished tasks :) later."

**Comment by Aritma** [comment]
Obsolete - never implemented

**Pull request closed without merging**